proposals/0000-package-dependency-import-syntax.md

```diff
@@ -0,0 +1,154 @@
+* Authors: [Rahul Malik](https://github.com/rahul-malik), [Ankit Aggarwal](https://github.com/aciidb0mb3r), [David Hart](https://github.com/hartbit)
+* Review Manager: TBD
+* Status: **Awaiting implementation**
+
+## Introduction
+
+Swift is a general-purpose language that aims expand it's availability and impact on various domains and platforms. We believe great scripting support and experience is an important part of improving the impact of Swift as a language. Swift already includes basic support for scripting via swift command-line tool. This is a proposal for greatly improving the script support by providing a deeper integration with the Swift Package Manager.
+
+
+Swift-evolution thread: [Discussion thread topic for that
+proposal](https://forums.swift.org/)
+
+## Motivation
+
+Swift standard library maintains a "batteries-not-included" approach and relies on external packages to provide more functionality. However, this introduces friction in writing Swift scripts as there's no easy way to reference external packages from a single file. For example, there's no way to easily include basic scripting utilities like a command-line argument parsing library.
+
+The Swift community has also recognized and tried to provide better scripting support through projects like Marathon and swift-sh. These projects leverage the Swift Package Manager to fetch and build external packages referenced in a Swift script. We believe we can take this approach further and provide a much better scripting experience by improving the integration between the Swift language and the Swift Package Manager. This will also allow Swift users to easily share their scripts without requiring to install additional tooling.
+
+This proposal would dramatically improve the developer experience for writing scripts in Swift. Dependencies for common use-cases would be easy to integrate and scripts would be portable across machines without additional tools installed on the host machine. 
+
+## Proposed solution
+
+We propose to add a new attribute @package to the import statement for referencing Swift packages. This attribute would be only valid in Swift script file and will be rejected if used during compilation of a regular Swift module. Swift Package Manager will then fetch and build the declared packages and make them available to the script when executing it.
+
+```
+import Foundation
+
+@package(url: "https://github.com/jpsim/Yams.git", from: "2.0.0")
+import Yams
+```
+
+This script can be executable by using `swift run MyScript.swift` which will utilize Swift package managers dependency fetching and resolution logic.
+
+The `@package(...)` syntax will contain the same level of control and specificity as SwiftPM current [Package Dependency Description](https://docs.swift.org/package-manager/PackageDescription/PackageDescription.html#package-dependency) which should feel like a natural to developers familiar with Swift Package Manager manifest syntax.  
```

Comment:
**comment:** Purely stylistically, perhaps instead of saying "syntax will contain the same level of control and specificity as SwiftPM current Package Dependency Description", we could be even more precise and say "syntax will allow the same parameters as SwiftPM's Package Dependency Description".
**typo:** should feel ~~like a natural~~ to developers